File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day23CoprocessorConflagrationTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import advent.twenty_seventeen.support.DuetCodeGenerator;
import org.junit.Ignore;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day23CoprocessorConflagrationTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day23CoprocessorConflagration("/twenty_seventeen/Day23-CoprocessorConflagration-input.txt");

        assertEquals(3025, dp.getPart1Answer().intValue());
        assertEquals(915, dp.getPart2Answer().intValue());
    }

    @Test
    @Ignore
    public void generateProgram() {
        DuetCodeGenerator generator = new DuetCodeGenerator("/twenty_seventeen/Day23-CoprocessorConflagration-input.txt");
        generator.generate(System.out);
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day23CoprocessorConflagrationTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day23CoprocessorConflagrationTest  extends FlatSpec with Matchers {
  "2017 Day 23 - Input" should "calculate answers" in {
    val dp = new Day23CoprocessorConflagration("/twenty_seventeen/Day23-CoprocessorConflagration-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day23CoprocessorConflagration("/twenty_seventeen/Day23-CoprocessorConflagration-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
